server/server.go

```diff
@@ -50,7 +50,9 @@ func grpcHandlerFunc(ctx context.Context, grpcServer *grpc.Server, otherHandler
 }
 
 // initHTTPServer initializes HTTP server with TLS credentials and returns http.Server.
-func initHTTPServer(ctx context.Context, tlsConfig *tls.Config, grpcServer *grpc.Server, gwmux *runtime.ServeMux, addr string) *http.Server {
+func initHTTPServer(ctx context.Context, tlsConfig *tls.Config,
+	grpcServer *grpc.Server, gwmux *runtime.ServeMux, addr string,
```

Comment:
Can we pass in a struct instead of multiple arguments. It would be easy to extend and readable.

Reply:
I thought about it, and I decided not to do so due to the following reasons:

- If we're passing a struct here, then it's better to also have a corresponding struct in `Config`. However, comparing to `ShutdownOnInternalFailureCriteria`, it seems that unnecessary complexity will be introduced if an additional grouping is added for these fields (maybe `HTTPServerTimeouts`, but that also causes stutters).
- The fields should not change often because they are mapped to the fields of `http.Server`.